spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java

```diff
@@ -303,6 +310,17 @@ public void testNetClientAndServerSSL() throws Exception {
 
 	@Test
 	public void testNetClientAndServerSSLDifferentContexts() throws Exception {
+		testNetClientAndServerSSLDifferentContexts(false);
+		try {
+			testNetClientAndServerSSLDifferentContexts(true);
+			fail("expected Exception");
+		}
+		catch (SSLHandshakeException e) {
```

Comment:
This passes for me or fails sporadically 😢.
With the :
```
java.net.SocketException: Software caused connection abort: recv failed

	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:170)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	at sun.security.ssl.SSLSocketImpl.waitForClose(SSLSocketImpl.java:1769)
	at sun.security.ssl.HandshakeOutStream.flush(HandshakeOutStream.java:124)
	at sun.security.ssl.Handshaker.sendChangeCipherSpec(Handshaker.java:1083)
	at sun.security.ssl.ClientHandshaker.sendChangeCipherAndFinish(ClientHandshaker.java:1222)
	at sun.security.ssl.ClientHandshaker.serverHelloDone(ClientHandshaker.java:1134)
	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:348)
	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:979)
	at sun.security.ssl.Handshaker.process_record(Handshaker.java:914)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062)
	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375)
	at sun.security.ssl.SSLSocketImpl.writeRecord(SSLSocketImpl.java:747)
	at sun.security.ssl.AppOutputStream.write(AppOutputStream.java:123)
	at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
	at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
	at org.springframework.integration.ip.tcp.connection.TcpNetConnection.send(TcpNetConnection.java:107)
	at org.springframework.integration.ip.tcp.connection.SocketSupportTests.testNetClientAndServerSSLDifferentContexts(SocketSupportTests.java:361)
	at org.springframework.integration.ip.tcp.connection.SocketSupportTests.testNetClientAndServerSSLDifferentContexts(SocketSupportTests.java:315)
```
or 
```
javax.net.ssl.SSLException: java.net.SocketException: Software caused connection abort: recv failed

	at sun.security.ssl.Alerts.getSSLException(Alerts.java:208)
	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949)
	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1906)
	at sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1870)
	at sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1815)
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:116)
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:71)
	at org.springframework.integration.ip.tcp.serializer.ByteArrayCrLfSerializer.fillToCrLf(ByteArrayCrLfSerializer.java:60)
	at org.springframework.integration.ip.tcp.serializer.ByteArrayCrLfSerializer.doDeserialize(ByteArrayCrLfSerializer.java:48)
	at org.springframework.integration.ip.tcp.serializer.AbstractPooledBufferByteArraySerializer.deserialize(AbstractPooledBufferByteArraySerializer.java:82)
	at org.springframework.integration.ip.tcp.serializer.AbstractPooledBufferByteArraySerializer.deserialize(AbstractPooledBufferByteArraySerializer.java:34)
	at org.springframework.integration.ip.tcp.connection.TcpNetConnection.getPayload(TcpNetConnection.java:121)
	at org.springframework.integration.ip.tcp.connection.TcpMessageMapper.toMessage(TcpMessageMapper.java:154)
	at org.springframework.integration.ip.tcp.connection.TcpNetConnection.run(TcpNetConnection.java:165)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.SocketException: Software caused connection abort: recv failed
```
Yes, on Windows.
I wonder if we still have some race condition somewhere ...